File: c/exemples/exemple_graphe_liste_adjacence.c
#include "graphe_liste_adjacence.h"

#include <stdio.h>

int main(void) {
    GrapheListeAdjacence graphe = creer_graphe_liste_adjacence(4, 1);

    ajouter_arete_graphe_liste_adjacence(&graphe, 0, 1);
    ajouter_arete_graphe_liste_adjacence(&graphe, 0, 2);
    ajouter_arete_graphe_liste_adjacence(&graphe, 2, 3);

    printf("Listes d'adjacence initiales :\n");
    afficher_graphe_liste_adjacence(&graphe);

    printf("L'arete 0 -> 2 existe : %s\n",
        arete_existe_graphe_liste_adjacence(&graphe, 0, 2) ? "oui" : "non");

    retirer_arete_graphe_liste_adjacence(&graphe, 0, 2);
    printf("Apres suppression de l'arete 0 -> 2 :\n");
    afficher_graphe_liste_adjacence(&graphe);

    detruire_graphe_liste_adjacence(&graphe);
    return 0;
}

File: ocaml/exemples/exemple_graphe_liste_adjacence.ml
let chaine_de_liste_entiere valeurs =
  "[" ^ String.concat " ; " (List.map string_of_int valeurs) ^ "]"

let () =
  let graphe = Graphe_liste_adjacence.creer_graphe_liste_adjacence 4 true in
  Graphe_liste_adjacence.ajouter_arete_graphe_liste_adjacence graphe 0 1;
  Graphe_liste_adjacence.ajouter_arete_graphe_liste_adjacence graphe 0 2;
  Graphe_liste_adjacence.ajouter_arete_graphe_liste_adjacence graphe 2 3;
  Printf.printf "Voisins de 0 : %s\n"
    (chaine_de_liste_entiere
       (Graphe_liste_adjacence.voisins_graphe_liste_adjacence graphe 0));
  Printf.printf "L'arete 0 -> 2 existe : %b\n"
    (Graphe_liste_adjacence.arete_existe_graphe_liste_adjacence graphe 0 2);
  Graphe_liste_adjacence.retirer_arete_graphe_liste_adjacence graphe 0 2;
  Printf.printf "Voisins de 0 apres suppression : %s\n"
    (chaine_de_liste_entiere
       (Graphe_liste_adjacence.voisins_graphe_liste_adjacence graphe 0))
